File: Python/problem69.py
def phi(x):
	ret = 1
	i = 2
	while x != 1:
		i = i+1
		pow = 1
		while not(x % i):
			x /= i
			pow *= i
		ret *= pow - pow/i
	return ret

def main():
	maximum = 0
	n = 0
	for i in range(2,1000):
		ratio = i / phi(i)
		if ratio > maximum:
			maximum = ratio
			n = i
			print n
	print n
	
if __name__ == "__main__":
	main()

File: problem69.cpp
#include <iostream>
#include <math.h>

using namespace std;

int phi(unsigned long x) {
	unsigned int ret = 1,i,pow;
	for (i = 2; x != 1; i++) {
		pow = 1;
		while (!(x%i)) {
			x /= i;
			pow *= i;
		}
		ret *= (pow - (pow/i));
	}
	return ret;
}


int main() {
	int highn = 0;
	double f = 0;
	for (int i=2; i<=1000000; i++) {
		int q = phi(i);
		if ((double)i/q > f) {
			f = (double)i/q;
			highn = i;
			cout << i << endl;
		}
	}
	cout << highn;
	return 0;
}

